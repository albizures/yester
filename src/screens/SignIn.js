import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Text, View, Button } from 'react-native'
// import colors from '../utils/colors'
import CustomTextInput from '../components/CustomTextInput'

export default class SignIn extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  onPress = () => {
    this.props.navigation.navigate('Home')
  }

  render () {
    return (
      <View style={{alignItems: 'center'}}>
        <Text style={[{textAlign: 'center', marginTop: 40}]}>SIGN IN</Text>
        <Button title='to Home' onPress={this.onPress} />
        <CustomTextInput title='signin.firstName' />
      </View>
    )
  }
}
