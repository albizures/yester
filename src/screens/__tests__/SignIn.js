import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import SignIn from '../SignIn'

const navigation = { navigate: jest.fn() }

describe.skip('src/screens/SignIn.js', () => {
  it('should render', () => {
    const tree = renderer.create(<SignIn navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to home is pressed', () => {
    it('should navigate to Home screen', () => {
      const shallowSignIn = shallow(<SignIn navigation={navigation} />)
      const shallowTouchableHighlights = shallowSignIn.find(Button)
      const shallowButtonToHome = shallowTouchableHighlights.at(0)

      shallowButtonToHome.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Home')
    })
  })
})
