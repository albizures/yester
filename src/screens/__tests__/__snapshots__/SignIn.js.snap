// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/SignIn.js should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <RCTScrollView

    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "marginTop": 40,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign in
      </Text>
      <Text
        style={
          Array [
            Object {
              "marginTop": 40,
              "textAlign": "center",
            },
          ]
        }
      >
        Starting this journey is simpler, fast and secure!
      </Text>
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          secureTextEntry={true}
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#4241B3",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
